### PR TITLE
Add more complex canvas reftest

### PR DESCRIPTION
**Pull request by kainino0x**
~dependent on #164~

### CODE REVIEW

**Comment by kainino0x** [comment]
@shaoboyan PTAL!

**Comment by shaoboyan** [comment]
@kainino0x  LGTM(And it seems that this will cover the orientation too).  Is this tested under wpt?

File: src/webgpu/web-platform/reftests/canvas_complex.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    usage: GPUTextureUsage.COPY_DST,
+  });
+
+  const rows = 2;
+  const bytesPerRow = 256;
+  const [buffer, mapping] = t.device.createBufferMapped({
+    size: rows * bytesPerRow,
+    usage: GPUBufferUsage.COPY_SRC,
+  });
+  switch (format) {
```
Comment (shaoboyan) [inline src/webgpu/web-platform/reftests/canvas_complex.ts:35]: A quick(silly) question, No rgba8unorm?

Comment (kainino0x) [reply src/webgpu/web-platform/reftests/canvas_complex.ts:35]: That's right. Initially I did test it, but actually I decided to disallow it in Chrome, as it's not supposed to be supported by default. (bgra8unorm and rgba16float were the two formats we tentatively agreed on for core.)

**Comment by kainino0x** [comment]
Thanks for the review!

**Pull request merged**